kyuubi-server/src/test/scala/org/apache/kyuubi/events/handler/ServerKafkaLoggingEventHandlerSuite.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.events.handler
+
+import java.net.ServerSocket
+import java.time.Duration
+import java.util.Properties
+
+import scala.collection.JavaConverters._
+import scala.util.Random
+
+import com.fasterxml.jackson.databind.json.JsonMapper
+import kafka.server.{KafkaConfig, KafkaServer}
+import org.apache.kafka.clients.admin.{AdminClient, NewTopic}
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.common.utils.Time
+
+import org.apache.kyuubi._
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.events.handler.ServerKafkaLoggingEventHandler.KAFKA_SERVER_EVENT_HANDLER_PREFIX
+import org.apache.kyuubi.operation.HiveJDBCTestHelper
+import org.apache.kyuubi.zookeeper.{EmbeddedZookeeper, ZookeeperConf}
+
+class ServerKafkaLoggingEventHandlerSuite extends WithKyuubiServer with HiveJDBCTestHelper
```

Comment:
suggest to use Testcontianers, and move it to a dedicated it module

Reply:
Also, there is `Mock*` in Kafka client, which is much light that launch real Kafka broker

Reply:
I may need further investigation for mocking a Kafka broker. I also considered the TestingServer in `curator-test`, as the way does in Ranger, but it's not working since the crossed versions among maven dependencies.

Reply:
mark: https://www.baeldung.com/kafka-mockproducer

Reply:
MockProducer of `Mock*` is not enough for testing which requires simulating a kafka server with bootstrap server URL. Turn to use TestContainers for staring Kafka instance with Docker image.